File: java-app/src/main/java/com/dexter/utils/Validator.java
package com.dexter.utils;

import android.content.Context;
import android.support.annotation.StringRes;
import android.widget.EditText;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Validator {

    /**
     * A method to validate an editText with a regex
     *
     * @param context          the context of the class to access class methods
     * @param editText         the editText which is to be validated
     * @param regex            the regular expression for which the validation is needed
     * @param errorStringResId the string to be shown when editText is not validated
     * @return true if is validate else false
     */
    public static boolean isValidate(Context context, EditText editText, String regex, @StringRes int errorStringResId) {
        String text = editText.getText().toString().trim();
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(text);
        if (!matcher.matches()) {
            editText.requestFocus();
            editText.setError(context.getString(errorStringResId));
            return false;
        } else {
            editText.setError(null);
            return true;
        }
    }
}

File: kotlin-app/src/main/java/com/dexter/utils/Validator.kt
package com.dexter.utils

import android.content.Context
import android.support.annotation.StringRes
import android.widget.EditText
import java.util.regex.Pattern

object Validator {

    /**
     * A method to validate an editText with a regex
     *
     * @param context          the context of the class to access class methods
     * @param editText         the editText which is to be validated
     * @param regex            the regular expression for which the validation is needed
     * @param errorStringResId the string to be shown when editText is not validated
     * @return true if is validate else false
     */
    fun isValidate(context: Context, editText: EditText, regex: String, @StringRes errorStringResId: Int): Boolean {
        val text = editText.text.toString().trim()
        val pattern = Pattern.compile(regex)
        val matcher = pattern.matcher(text)
        return if (!matcher.matches()) {
            editText.requestFocus()
            editText.error = context.getString(errorStringResId)
            false
        } else {
            editText.error = null
            true
        }
    }
}
